lib/DiscoveryStreamFeed.jsm

```diff
@@ -353,9 +353,15 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
       (this.store.getState().Prefs.values[PREF_SPOCS_ENDPOINT] ||
         this.config.spocs_endpoint)
     ) {
-      layout.spocs.url =
-        this.store.getState().Prefs.values[PREF_SPOCS_ENDPOINT] ||
-        this.config.spocs_endpoint;
```

Comment:
Alternatively, you can leave this code untouched and instead remove the global `basicLayoutResp` and `defaultLayoutResp` and do something like:
```js
 layout = { lastUpdate: Date.now(), ...getHardcoded("basic") };

function getHardcoded(which) {
  return which === "basic" ? { …the definition for basic layout goes here… } : { … };
}
```

So `getHardcoded()` always returns a new js object similar to how `fetchLayout` returns a new js object (from JSON.parse). Or I suppose maybe another alternative is make basicLayoutResp as a json string that gets parsed to create a new js object?

These other approaches might be safer if some future change always accidentally wants to change the original hardcoded layout object.

Reply:
Hm, actually now that I think about it.

1. We don't need to change the `spocs.url`, just when we fire `DISCOVERY_STREAM_SPOCS_ENDPOINT`we send the right thing, the rest of the code interacting with layout in this file should be looking at what's in redux, not the layout object.
2. With the above, further changes to the hard coded object like I did should be avoided. I think maybe a comment about not modifying the layouts is probably enough at that point.

Reply:
Trying a much simpler solution form my first one, but not sure how you think it compares to your ideas, thoughts?